gui/widgets/editable.cpp

```diff
@@ -63,20 +63,32 @@ void EditableWidget::reflowLayout() {
 }
 
 void EditableWidget::setEditString(const String &str) {
-	// TODO: We probably should filter the input string here,
-	// e.g. using tryInsertChar.
-	_editString = str;
+	_editString = "";
+
+	// Copy string char by char, ensuring each char is valid
+	uint len = 0;
+	for (uint i = 0; i < str.size() && len < _maxLength; i++)
+		if (isCharValid((byte)str[i])) {
+			_editString += str[i];
+			len++;
+		}
+
 	_caretPos = 0;
 }
 
 bool EditableWidget::tryInsertChar(byte c, int pos) {
-	if ((c >= 32 && c <= 127) || c >= 160) {
+	if (_maxLength == 0 || _editString.size() < _maxLength) {
```

Comment:
We have at least one subclass of EditableWidget which overrides tryInsertChar to do custom filtering of input, it can be found over [here](https://github.com/scummvm/scummvm/blob/master/gui/launcher.cpp#L99-L104).

They way I see it this overriding will now break the length checking you added.

It is also not clear to me why you don't call `isCharValid` inside `tryInsertChar`. Which seems the obvious place to do it. Then you could also simply use `tryInsertChar` in `setEditString`.